pkg/runtime/scheduler/scheduler.go

```
@@ -0,0 +1,173 @@
+		})
+	}
+
+	return manager, nil
+}
+
+// Run starts watching for job triggers from all scheduler clients.
+func (m *Manager) Run(ctx context.Context) {
+	m.wg.Add(len(m.Clients))
+
+	// Start a goroutine for each client to watch for job updates
+	for _, client := range m.Clients {
+		go func(client *Client) {
+			defer m.wg.Done()
+			m.watchJob(ctx, client)
+		}(client)
+	}
+}
+
+// watchJob starts watching for job triggers from a single scheduler client.
+func (m *Manager) watchJob(ctx context.Context, client *Client) {
+	streamReq := &schedulerv1pb.StreamJobRequest{
+		AppId:     m.ConnDetails.AppID,
+		Namespace: m.ConnDetails.Namespace,
+		Hostname:  m.ConnDetails.Host,
+		Port:      int32(m.ConnDetails.Port),
+	}
+
+	defer func() {
+		if err := client.conn.Close(); err != nil {
+			log.Errorf("Error closing connection: %v", err)
+		}
+	}()
+
+	backoffPolicy := backoff.NewExponentialBackOff()
+	backoffPolicy.MaxElapsedTime = 0 // Retry indefinitely
+	backoffPolicy.InitialInterval = 30 * time.Second
+	backoffPolicy.MaxInterval = 5 * time.Minute
+
+	err := backoff.Retry(func() error {
+		select {
+		case <-ctx.Done():
+			log.Infof("Context cancelled. Exiting watchJob goroutine.")
+			return backoff.Permanent(ctx.Err())
+		default:
+			stream, err := client.scheduler.WatchJob(ctx, streamReq)
+			if err != nil {
+				log.Errorf("Error while streaming with Scheduler: %v. Retrying...\n", err)
+				return err // retryable error
+			}
+
+			for {
+				select {
+				case <-ctx.Done():
+					log.Infof("Context cancelled. Exiting watchJob goroutine.")
+					return nil
+				default:
+					resp, err := stream.Recv()
+					if err != nil {
+						log.Errorf("Error while receiving job triggers: %v. Retrying...\n", err)
+						return err // retryable error
+					}
+					log.Infof("Received response: %v", resp) // TODO: Send resp back to apps
+				}
+			}
+		}
+	}, backoffPolicy)
+
+	if err != nil {
+		log.Errorf("Error from backoff retry to Scheduler. Likely ctx cancelled.")
```

This should return an error.